In [1]:
import numpy as np

## 1️⃣ zeros()

1. Create a 2x3 zeros matrix.

2. Create a 4x4 zeros matrix of type int.

In [3]:
arr = np.zeros((2, 3))
arr

array([[0., 0., 0.],
       [0., 0., 0.]])

In [5]:
arr1 = np.zeros((4, 4), dtype=int)
arr1

array([[0, 0, 0, 0],
       [0, 0, 0, 0],
       [0, 0, 0, 0],
       [0, 0, 0, 0]])

## 2️⃣ ones()

3. Create a 3x2 ones matrix.

4. Create a 5x5 ones matrix and multiply it by 7.

In [7]:
arr = np.ones((3, 2))
arr

array([[1., 1.],
       [1., 1.],
       [1., 1.]])

In [12]:
arr1 =np.ones((5, 5)) * 7
arr1

array([[7., 7., 7., 7., 7.],
       [7., 7., 7., 7., 7.],
       [7., 7., 7., 7., 7.],
       [7., 7., 7., 7., 7.],
       [7., 7., 7., 7., 7.]])

## 3️⃣ eye()
- `np.eye(N, M=None, k=0)`

5. Create a 3x3 identity matrix.

6. Create a 5x5 identity matrix with diagonal offset k=1.

In [ ]:
arr = np.eye(3)
arr

array([[1., 0., 0.],
       [0., 1., 0.],
       [0., 0., 1.]])

In [17]:
arr1 = np.eye(5, k = 1)
arr1

array([[0., 1., 0., 0., 0.],
       [0., 0., 1., 0., 0.],
       [0., 0., 0., 1., 0.],
       [0., 0., 0., 0., 1.],
       [0., 0., 0., 0., 0.]])

## 4️⃣ arange()
- `np.arange(start, stop, step)`

7. Create an array of numbers from 0 to 9.

8. Create an array from 10 to 50 with step 5.

9. Create an array from -5 to 5 with step 0.5.

In [18]:
arr = np.arange(0, 10)
arr

array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])

In [19]:
arr1 = np.arange(10, 51, 5)
arr1

array([10, 15, 20, 25, 30, 35, 40, 45, 50])

In [21]:
arr2 = np.arange(-5, 5.5, 0.5)
arr2

array([-5. , -4.5, -4. , -3.5, -3. , -2.5, -2. , -1.5, -1. , -0.5,  0. ,
        0.5,  1. ,  1.5,  2. ,  2.5,  3. ,  3.5,  4. ,  4.5,  5. ])

## 5️⃣ linspace()
`np.linspace(start, stop, num)`
### Parameters:
- start → shuru ka number
- stop → last number
- num → total kitne numbers chahiye

10. Create 5 numbers equally spaced between 0 and 1.

11. Create 10 numbers equally spaced between 2 and 5.

12. Create 7 numbers between -3 and 3 and round them to 2 decimal places.

In [22]:
arr = np.linspace(0, 1, 5)
arr

array([0.  , 0.25, 0.5 , 0.75, 1.  ])

In [23]:
arr1 = np.linspace(2, 5, 10)
arr1

array([2.        , 2.33333333, 2.66666667, 3.        , 3.33333333,
       3.66666667, 4.        , 4.33333333, 4.66666667, 5.        ])

In [24]:
arr3 = np.linspace(-3, 3, 7)
arr = np.round(arr3, 2)
arr


array([-3., -2., -1.,  0.,  1.,  2.,  3.])

## 6️⃣ empty()
`np.empty(shape)`

13. Create a 2x3 empty array and observe the values.

14. Create a 4x4 empty array and explain why the values are random.

## 7️⃣ full()

15. Create a 3x3 matrix filled with 9.

16. Create a 4x5 matrix filled with -1.

17. Create a 2x6 matrix filled with 3.5.

# 8️⃣ Mixed Special Arrays Practice

18. Create a 3x3 identity matrix and add a 3x3 ones matrix to it.

19. Create a 5x5 zeros matrix and fill its diagonal with 7 using `np.fill_diagonal()`.

20. Create a 1D array of 10 numbers from 0 to 9 using `arange()` and then convert it into a 2x5 matrix using `reshape()`.